splunk/splunk-operator
# CSPL-957: Automation tests to verify deletion of C3 SVA & refactoring of test cases to align with the model of "CRUD of SVAs"

- Adding new test for the bug described CSPL-780:  
  - "crcrud: can verify IDXC, CM and SHC PVCs are correctly deleted after the CRs deletion"
  - Steps for this new test:
  - Deploy C3 SVA (CM,IDXC,SHC with deployer)
  - Once everything is ready, verify PVCs (etc and var) exists for CM,IDXC,SHC, deployer
  - Delete the SHC,IDXC,CM
  - verify PVCs (etc and var) have been deleted for CM,IDXC,SHC, deployer

 - Refactor of the 'custom_resource_crud' folder:
    -  Split test cases into individual files based on SVA (S1, C3, M4)

Tests runs:
New test:
![image](https://user-images.githubusercontent.com/9527069/113228545-f542dd80-9249-11eb-83e3-4f96864dc6dd.png)

Refactored tests:
![image](https://user-images.githubusercontent.com/9527069/113233467-22948900-9254-11eb-814c-8a11e507d561.png)
![image](https://user-images.githubusercontent.com/9527069/113233487-2aecc400-9254-11eb-878d-ff99ebc552e2.png)
![image](https://user-images.githubusercontent.com/9527069/113233498-317b3b80-9254-11eb-89f8-8282040ca74d.png)
